tests/ui/lint/non-snake-case/lint-non-snake-case-crate.rs

```diff
@@ -0,0 +1,26 @@
+//! Don't lint on binary crate with non-snake-case names.
+//!
+//! See <https://github.com/rust-lang/rust/issues/45127>.
+
+//@ revisions: bin_ cdylib_ dylib_ lib_ proc_macro_ rlib_ staticlib_
```

Comment:
```suggestion
//@ revisions: bin cdylib dylib lib proc_macro rlib staticlib
```
Why can't the versions without underscores be used for revisions?

Reply:
E.g. `proc_macro` is a built-in cfg

```
error: unexpected `--cfg proc_macro` flag
   |
   = note: config `proc_macro` is only supposed to be controlled by `--crate-type proc-macro`
   = note: manually setting a built-in cfg can and does create incoherent behaviors
   = note: `#[deny(explicit_builtin_cfgs_in_flags)]` on by default
```

I just added a trailing `_` to all of them to be consistent.

Reply:
@rustbot ready